### PR TITLE
SIG-1647 Fix existing PHP verifications and add new verifications

### PR DESCRIPTION
**Pull request by ATimmeh33**

### CODE REVIEW

File: signhost.php [hunk 1]
```diff
@@ -336,6 +336,116 @@ function jsonSerialize() {
 
 }
 
+class DigidVerification extends Verification implements JsonSerializable {
+
+    public $Bsn; // String
```
Comment (MrJoe) [inline signhost.php]: Wrong indent

File: signhost.php [hunk 2]
```diff
@@ -336,6 +336,116 @@ function jsonSerialize() {
...
+
+	function jsonSerialize() {
+		return array("Type" => $this->Type);
+	}
+
+}
+
+class ScribbleVerification extends Verification implements JsonSerializable {
+
+    public $RequireHandsignature; // Bool
```
Comment (MrJoe) [inline signhost.php]: Wrong indent

File: signhost.php [hunk 3]
```diff
@@ -336,6 +336,116 @@ function jsonSerialize() {
...
+			"RequireHandsignature" => $this->RequireHandsignature,
+			"ScribbleNameFixed" => $this->ScribbleNameFixed,
+			"ScribbleName", $this->ScribbleName);
+	}
+
+}
+
+class PhoneNumberVerification extends Verification implements JsonSerializable {
+
+    public $Number; // String
```
Comment (MrJoe) [inline signhost.php]: Wrong indent

**Review by MrJoe — CHANGES_REQUESTED** [review]

File: signhost.php [hunk 4]
```diff
@@ -293,30 +285,157 @@ function __construct($type) {
 
 }
 
-class iDEAL extends Verification {
+class IdealVerification extends Verification implements JsonSerializable {
 
     public $Iban; // String
```
Comment (MrJoe) [inline signhost.php]: Maybe document the properties with @var syntax?
https://stackoverflow.com/questions/2194822/how-to-document-class-properties-in-php-5-with-phpdocumentor#2194835
Don't if there are also tags for readonly and optional

**Review by MrJoe — CHANGES_REQUESTED** [review]

File: signhost.php [hunk 5]
```diff
@@ -187,28 +237,115 @@ function __construct(
...
+
+	/**
+	 * @var string
+	 */
+	public $BSN;
+
+	/**
+	 * @var bool
+	 */
+	public $RequireScribble;
```
Comment (MrJoe) [inline signhost.php]: Should be part of verifications array

Comment (MrJoe) [inline signhost.php]: Should be part of verifications array

File: signhost.php [hunk 6]
```diff
@@ -187,28 +237,115 @@ function __construct(
...
+
+	/**
+	 * @var bool
+	 */
+	public $RequireSmsVerification;
```
Comment (MrJoe) [inline signhost.php]: Should be part of verifications array

File: signhost.php [hunk 7]
```diff
@@ -187,28 +237,115 @@ function __construct(
...
+
+	/**
+	 * @var bool
+	 */
+	public $RequireDigidVerification;
```
Comment (MrJoe) [inline signhost.php]: Should be part of verifications array

File: signhost.php [hunk 8]
```diff
@@ -187,28 +237,115 @@ function __construct(
...
+
+	/**
+	 * @var bool
+	 */
+	public $RequireKennisnetVerification;
```
Comment (MrJoe) [inline signhost.php]: Should be part of verifications array

File: signhost.php [hunk 9]
```diff
@@ -187,28 +237,115 @@ function __construct(
...
+
+	/**
+	 * @var bool
+	 */
+	public $RequireSurfnetVerification;
```
Comment (MrJoe) [inline signhost.php]: Should be part of verifications array

File: signhost.php [hunk 10]
```diff
@@ -187,28 +237,115 @@ function __construct(
...
+
+	/**
+	 * @var string
+	 */
+	public $Language;
+
+	/**
+	 * @var string
+	 */
+	public $ScribbleName;
```
Comment (MrJoe) [inline signhost.php]: Should be part of verifications array

File: signhost.php [hunk 11]
```diff
@@ -187,28 +237,115 @@ function __construct(
...
+
+	/**
+	 * @var bool
+	 */
+	public $ScribbleNameFixed;
```
Comment (MrJoe) [inline signhost.php]: Should be part of verifications array

**Review by MrJoe — CHANGES_REQUESTED** [review]
Old verification properties should be removed

**Comment by ATimmeh33** [comment]
4350286c02712be9f3010cf2eefdd759cd3f4b2c is no longer needed, already done in https://github.com/Evidos/signhost-phpdemoclient/pull/18

**Comment by ATimmeh33** [comment]
Obsoleted by https://github.com/Evidos/signhost-phpdemoclient/pull/21

**Pull request closed without merging**